vllm/worker/model_runner.py

```diff
@@ -1047,29 +1047,27 @@ def capture(
         # Run the model once without capturing the graph.
         # This is to make sure that the captured graph does not include the
         # kernel launches for initial benchmarking (e.g., Triton autotune).
-        with graph_mode():
-            self.model(
-                input_ids,
-                positions,
-                kv_caches,
-                attn_metadata,
-                **kwargs,
-            )
+        self.model(
+            input_ids,
+            positions,
+            kv_caches,
+            attn_metadata,
+            **kwargs,
+        )
         torch.cuda.synchronize()
 
         # Capture the graph.
         # NOTE(woosuk): Python 3.8 does not support multi-line with statements.
         # https://stackoverflow.com/questions/31039022/python-multi-line-with-statement
         self._graph = torch.cuda.CUDAGraph()
         with torch.cuda.graph(self._graph, pool=memory_pool):  # noqa: SIM117
-            with graph_mode():
```

Comment:
I just recalled why I started the graph mode inside `torch.cuda.graph`.

IIUC, `torch.cuda.graph` (without the input `stream` specified) starts a new CUDA stream in order to explicitly distinguish the kernels to capture from other kernels possibly launched on background. Thus, we need to either set the stream of `pynccl` after entering `torch.cuda.graph` context or somehow manually make sure that pynccl and cuda graph uses the same stream.

Reply:
Nice catch! Will address this later.

Reply:
[1a0721c](https://github.com/vllm-project/vllm/pull/4818/commits/1a0721c20c4404d83d640394c3108eacef0a89f3) should address this issue.